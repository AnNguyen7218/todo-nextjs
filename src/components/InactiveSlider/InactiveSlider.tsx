import React from 'react';
import {
  ReactIdSwiperProps,
  ReactIdSwiperChildren,
  GetSwiper,
} from 'react-id-swiper/lib/types';
import { Slider } from 'components';

const swiperDefaultParams: ReactIdSwiperProps = {
  autoplay: {
    delay: 800,
  },
  centeredSlides: true,
  speed: 1000,
  allowTouchMove: false,
};

type Props = {
  swiperParams?: ReactIdSwiperProps;
  children: ReactIdSwiperChildren;
  getSwiper?: GetSwiper;
};

const InactiveSlider: React.FC<Props> = ({
  swiperParams = {},
  children,
  getSwiper,
}) => {
  const mergedSwiperParams = { ...swiperDefaultParams, ...swiperParams };
  return (
    <Slider getSwiper={getSwiper} swiperParams={mergedSwiperParams}>
      {children}
    </Slider>
  );
};

export default InactiveSlider;
