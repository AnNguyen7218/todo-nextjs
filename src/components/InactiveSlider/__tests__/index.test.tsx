import React from 'react';
import { render, cleanup } from '@testing-library/react';
import InactiveSlider from '../InactiveSlider';

afterEach(cleanup);

describe('InactiveSlider Component', () => {
  test('InactiveSlider should match snapshot', () => {
    const { container } = render(
      <InactiveSlider>
        <div>Slide a</div>
        <div>Slide b</div>
        <div>Slide c</div>
        <div>Slide d</div>
      </InactiveSlider>
    );
    expect(container).toMatchSnapshot();
  });
});
