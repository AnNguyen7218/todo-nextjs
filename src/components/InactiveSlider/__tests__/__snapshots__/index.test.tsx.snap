// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InactiveSlider Component InactiveSlider should match snapshot 1`] = `
<div>
  <div
    class="swiper-container swiper-container-initialized swiper-container-horizontal"
  >
    <div
      class="swiper-wrapper"
      style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);"
    >
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="0"
      >
        Slide a
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="1"
      >
        Slide b
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="2"
      >
        Slide c
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="3"
      >
        Slide d
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="0"
      >
        Slide a
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="1"
      >
        Slide b
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="2"
      >
        Slide c
      </div>
      <div
        class="swiper-slide"
        data-swiper-slide-index="3"
      >
        Slide d
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="0"
      >
        Slide a
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="1"
      >
        Slide b
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="2"
      >
        Slide c
      </div>
      <div
        class="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index="3"
      >
        Slide d
      </div>
    </div>
    <span
      aria-atomic="true"
      aria-live="assertive"
      class="swiper-notification"
    />
  </div>
</div>
`;
